.github/workflows/build_pr.yml

```diff
@@ -0,0 +1,25 @@
+name: Build for PR
+
+on:
+  pull_request:
+    types: [ opened, synchronize ]
```

Comment:
In `Test Pull Request Changes` workflow we have:

```
[opened, synchronize, edited, ready_for_review]
```

Maybe it is worth adding `edited` also. The negative is that this workflow will also run again if the description or title is updated but the positive is that it will run if the target branch changes which I think we want.

Reply:
It's heavy and time consuming work that produces new builds. I'm not sure it would be worse if I added `edited` and `ready_for_review`

Reply:
True. I assume it doesn't cost _us_ anything (our CI time is free for GitHub right?) but isn't great for the planet and might be annoying having to wait for a rebuild if we have just edited the title!

Changing the target branch is probably a rare case so I imagine excluding `edited` if fine. It's just something for us to be aware of. I think we would only need `ready_for_review` if we kept `github.event.pull_request.draft == false` which we are not.

Reply:
it would cancel the current run anyway, editing the target branch does happen, I would add it.

Concerning the time to build, do we need to run these tests each time, I am thinking we probably don't want to run them before it has been reviewed. We could check for that it's approved or use a label to check to trigger the tests. what do you think @KaterinaWire ?

Reply:
> [opened, synchronize, edited, ready_for_review]

@samwyndham  added

Reply:
> it would cancel the current run anyway, editing the target branch does happen, I would add it.
> 
> Concerning the time to build, do we need to run these tests each time, I am thinking we probably don't want to run them before it has been reviewed. We could check for that it's approved or use a label to check to trigger the tests. what do you think @KaterinaWire ?

discussed this in a call